src/tiled/mainwindow.cpp

```diff
@@ -99,6 +102,95 @@ using namespace Tiled::Internal;
 using namespace Tiled::Utils;
 
 
+namespace {
+
+template <typename Format>
+struct ExportDetails
+{
+   Format* mFormat;
+   QString mFileName;
+
+   ExportDetails() {}
+   ExportDetails(Format* format, const QString& fileName)
+     : mFormat(format)
+     , mFileName(fileName)
+   {}
+
+   bool valid() { return static_cast<bool>(mFormat); }
```

Comment:
Naming style: please call this `isValid`. For implementation I prefer `return mFormat != nullptr;`

Reply:
Done